### ISSUE TITLE
[Features Request] Subtitles support + Overall MKV support

### ISSUE THREAD

**Issue by enzomtpYT** [sylviiu/ezytdl#56]
Hi ! I've been struggling while downloading some videos and i think some useful features could be added :
- Downloads of Subtitles
- Adding MKV support
- Downloading all audio tracks (from different languages) in one MKV
- Downloading all Subtitles (from different languages) in one MKV

I am sorry if i am asking too much, ill try to find some way to make what i want to do.

Thanks for make a such great program !

**Comment by sylviiu** [comment]
I would absolutely love to look into this! I've never had a use case personally for any of these but I'm willing to put in effort in implementing that :)

Which source are you trying to download from that supports multiple audio streams? I don't know of any personally.

**Comment by enzomtpYT** [comment]
Here is an example with multiple Audio Source & Multiple Subtitles : https://www.youtube.com/watch?v=mImFz8mkaHo

**Comment by enzomtpYT** [comment]
Also if this can help i found a way to download youtube subtitles by using YT-DL arguements : "--all-subs"

**Comment by sylviiu** [comment]
Noted, thank you. I'll look into this!

*i actually never knew youtube offered multiple language audio tracks*

Downloading with "--all-subs" would only fit with one use case (downloading with FFmpeg disabled, and downloading the original format without conversion -- which might not even embed the subtitles if possible) -- I'd have to find a way to embed it with FFmpeg as well. This can become a "metadata option" here:
![image](https://github.com/sylviiu/ezytdl/assets/95180094/9b1ede64-6e1b-4e37-a1e4-47d0fb02425d)

**Comment by sylviiu** [comment]
<details>
<summary>leaving this as a note for myself</summary>

multiple language streams are found in the formats array with the same format ID; e.g. format id 233 has multiple indexes for each languages (233-0, 233-1, etc), where the note specifies the language

![image](https://github.com/sylviiu/ezytdl/assets/95180094/ec9a90ec-5bf5-4beb-8bc1-a07117276e98)
![image](https://github.com/sylviiu/ezytdl/assets/95180094/dfa1013f-dc3b-4cbc-81f7-20490f600277)
![image](https://github.com/sylviiu/ezytdl/assets/95180094/00a46d9a-3af2-471c-adba-c0ab5423e7c0)

i can't tell if this is the same for all services, but crunchyroll languages are through different URLs.

</details>

**Comment by sylviiu** [comment]
Also, what do you mean by adding MKV support? What's missing from it?

All media encoding & processing is handled by FFmpeg (where applicable); MKV should be supported already?

**Comment by enzomtpYT** [comment]
Ok, wait till I get home and I will tell you more
Sorry for making you wait

**Comment by enzomtpYT** [comment]
> Also, what do you mean by adding MKV support? What's missing from it?
> 
> All media encoding & processing is handled by FFmpeg (where applicable); MKV should be supported already?

Okay so mkv is already supported by FFmpeg, but after checking mkv support multiple audio streams, i wanted to know if it is possible to make something where you choose what to put (with checkboxes) and then at the end it merge all audio/videos and subtitles in one mkv files with ffmpeg, sorry again if it is too difficult !

**Comment by sylviiu** [comment]
Yeah, that would be possible. That's currently my plan for implementing this.

Sorry for being silent past couple days, I've been a little busy outside of software development :( I'll work on this soon!

**Comment by enzomtpYT** [comment]
> Sorry for being silent past couple days, I've been a little busy outside of software development :(
It is okay, i don't want to rush you ! Take all the time you need !

**Comment by sylviiu** [comment]
finally got around to making some progress :D
![image](https://github.com/sylviiu/ezytdl/assets/95180094/6d070fca-cf43-475e-b935-db856d1cd40a)

**Comment by Another5405** [comment]
I need the Downloads of Subtitles function too, that will very helpful for my work.

**Comment by Gautam-Pra** [comment]
can you add multiple audio tracks function for https://www.discoveryplus.in/show/morgan-freemans-through-the-wormhole this website also?
